### PR TITLE
Add increment and decrement methods metrics

### PR DESCRIPTION
**Pull request by lucaslopezf**
### Description

This Pull Request introduces the capability to increment and decrement Gauge metrics. The key changes include:

1. Addition of a new interface, `IncrementDecrementHandler`, that extends the existing `MetricHandler` interface to include `Inc()` and `Dec()` methods.
2. Modification of `taskMetrics` to include `IncrementMetric` and `DecrementMetric` methods.
3. Elimination of code repetition in `taskMetrics` through refactoring.
4. Update of unit tests in `handler_test.go` to cover the new increment and decrement functionalities.

### CODE REVIEW

**Review by emmanuelm41 — APPROVED** [review]

File: pkg/metrics/handler.go [hunk 1]
```diff
@@ -1,25 +1,57 @@
...
 	if !ok {
-		zap.S().Errorf("Error: Metric not found for %s", name)
-		return
+		return fmt.Errorf("error: Metric not found for %s", name)
 	}
 
-	if err := metricDetail.Handler.Update(metricDetail.Collector, value, labels...); err != nil {
-		zap.S().Errorf("Error updating metric %s. Err: %s", name, err.Error())
+	if err := action(metricDetail.Handler, metricDetail.Collector, labels...); err != nil {
+		return fmt.Errorf("error performing action on metric %s. Err: %s", name, err.Error())
```
Comment (bizk) [inline pkg/metrics/handler.go:29]: why are we suing FMT instead of ZAP?

**Review by bizk — APPROVED** [review]
Seems good

File: pkg/metrics/handler.go (hunk 1 above)
Comment (lucaslopezf) [reply pkg/metrics/handler.go:29]: I use it to build the error, I prefer to return the error and have the client be the one to log it.

Comment (bizk) [reply pkg/metrics/handler.go:29]: sounds reasonable

**Pull request merged**